openedx/core/djangoapps/plugins/plugin_contexts.py

```diff
@@ -0,0 +1,35 @@
+from importlib import import_module
+
+from logging import getLogger
+from . import constants, registry
+
+log = getLogger(__name__)
+
+
+def get_plugins_view_context(project_type, view_name, existing_context={}):
```

Comment:
We will want to `memoize` the response of this call since it can be costly as it enumerates through all plugins and will always return the same value for an instantiation of the Django project.

Reply:
This is *super* important and I totally forgot about it.

Reply:
Just want to clarify that this entire thing can't be memoized. The plugin that I'm working on changes the context based on the user in the `existing_context`. The plugin checks if the user meets some criteria internal to the plugin, then adds additional context of `"<>_enabled": True/False`. I could cache everything up to actually calling the `context_function`, but that would have to stay uncached or we'd immediately fall over with out of memory errors. So I won't be able to use the nice `process_cache` decorator to do my work for me.

Reply:
That said, do you find that cache of everything up to the function, but not the function responses an acceptable middle ground? I'm not sure if that alleviates the performance concerns? When dashboard is loaded it would simply run the context through a series of predetermined functions which would add their context. No need for iterating, filtering, and loading.

Reply:
Good point.  I'm mostly suggesting memoizing [Lines 18-27](https://github.com/edx/edx-platform/pull/23330/files#diff-afc854f2f849b70198c561d761ba80b2R18-R27) where we retrieve the registered plugins in the system and call `import_module` on each.  You can memoize the list of registered plugins, but not memoize the responses of their `context_function`s.